pstoremem/addr_book.go

```diff
@@ -277,11 +277,6 @@ func (mab *memoryAddrBook) SetAddrs(p peer.ID, addrs []ma.Multiaddr, ttl time.Du
 			delete(amap, key)
 		}
 	}
-
-	// if we've expired all the signed addresses for a peer, remove their signed routing state record
-	if len(amap) == 0 {
-		delete(s.signedPeerRecords, p)
```

Comment:
We should actually switch this to `delete(s.addrs, p)`. Same below.

Reply:
@Stebalien 

Wont they be removed by GC ?

Reply:
Yes, but there's no reason not to clean this up early if we know it's empty.